composer/datasets/in_context_learning_evaluation.py

```diff
@@ -837,6 +848,228 @@ def collate_fn(self, data):
+            encoded_example['canonical_solution'] = self.samples[sample_idx]['canonical_solution']
+            encoded_example['test'] = self.samples[sample_idx]['test']
+            encoded_example['entry_point'] = self.samples[sample_idx]['entry_point']
+            encoded_example['test_inputs'] = self.samples[sample_idx]['test_inputs']
+            encoded_example['test_outputs'] = self.samples[sample_idx]['test_outputs']
+
+            examples.append(encoded_example)
+            max_prompt_length = max(
+                max_prompt_length,
+                len(encoded_example['preamble']['input_ids'] + encoded_example['prompt']['input_ids']))
+
+        self.max_prompt_length = max_prompt_length
+        return examples
+
+    def __getitem__(self, index):
+        return self.encoded_dataset[index]
+
+    def __len__(self):
+        return len(self.encoded_dataset)
+
+    def collate_fn(self, data):
+        inputs, prompts, tests, canonical_solutions, entry_points, test_inputs, test_outputs = [], [], [], [], [], [], []
+        for sample in data:
+            preamble, prompt, text_prompt, canonical_solution, test, entry_point, test_input, test_output = (
+                sample['preamble'], sample['prompt'], sample['prompt_text'], sample['canonical_solution'],
+                sample['test'], sample['entry_point'], sample['test_inputs'], sample['test_outputs'])
+            context_enc = preamble['input_ids'] + prompt['input_ids']
+            inp, _ = _make_padded_input(context_enc, [],
+                                        self.max_prompt_length,
+                                        self.pad_tok_id,
+                                        padding_side=self.padding_side)
+
+            inputs.append(inp)
+            tests.append(test)
+            prompts.append(text_prompt)
+            canonical_solutions.append(canonical_solution)
+            entry_points.append(entry_point)
+            test_inputs.append(test_input)
+            test_outputs.append(test_output)
+
+        batch = {
+            'input_ids': torch.stack(inputs),
+            'mode': 'generate',
+            'labels': canonical_solutions,
+            'prompts': prompts,  # list of prompts
+            'tests': tests,  # list of tests
+            'canonical_solutions': canonical_solutions,  # list of solutions
+            'entry_points': entry_points,  # list of entry points
+            'test_inputs': test_inputs,  # list of test inputs
+            'test_outputs': test_outputs,  # list of test outputs
+            'generation_length': self.max_seq_len - self.max_prompt_length,
+            'generation_kwargs': {
+                'pad_token_id': self.pad_tok_id,
+                'num_beams': self.num_evals,  # change strategy to beam search
+                'num_return_sequences': self.num_evals,  # how many gens per prompt
+                'do_sample': True,
+                'top_p': 0.95,
```

Comment:
Should these sampling params be hardcoded? I think they probably shouldn't be. Also, do we know if the humaneval results are sensitive to the sampling params? I think it would be a good idea to check that.

Reply:
We can check this in further depth

Reply:
done